Add bitcoin-core-26-0-and-f2pool-s-ofac-compliant-mining-policy review to bitcoin-explained (#655)
bitcointranscripts/bitcointranscripts: bitcoin-explained/bitcoin-core-26-0-and-f2pool-s-ofac-compliant-mining-policy.md

Aaron van Wirdum: 00:01:46

Yeah, so just to reiterate what this all means, there is an OFAC list and this, and this means there are entities on there that you're not allowed to transact with.
And that's the US government sanctioned list.
Is that the right way to think about it?

Sjors Provoost: 00:02:02

Yes, correct.

Aaron van Wirdum: 00:02:03

So these would include people that the US government does not like, essentially.

Sjors Provoost: 00:02:08

Yes, these are only people that the US government does not like.

Aaron van Wirdum: 00:02:11

Right, and there are Bitcoin addresses on this list?

Sjors Provoost: 00:02:14

Yeah, so I think the general idea is that they say, hey, this guy and this guy, you should not do business with them.
And this is how you recognize them.
And they'll spell out their name in like five different ways.
And give maybe some known addresses.
And they might also give some known Bitcoin addresses that are associated with this person.

Aaron van Wirdum: 00:02:31

Right.
So there are Bitcoin address on this list.

> And if you want to send Bitcoin to or from these addresses, then so what MarathonPool (MaraPool) was doing was they would essentially filter these transactions and not include these transactions into a block.

Reviewer: ```suggestion
And if you want to send Bitcoin to or from these addresses, then so what Marathon Pool (MaraPool) was doing was they would essentially filter these transactions and not include these transactions into a block.
```

Reviewer: You could click `apply suggestions` and `commit`.

[Applying suggested changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes)

That's how they would comply with this list, right?

Sjors Provoost: 00:02:49

Yeah, but they didn't actually do that.
I think they were thinking about doing that.

> And they put some operator message in the block.

Reviewer: ```suggestion
And they put some arbitrary message in the block.
```

Reviewer: <img width="2014" height="300" alt="image" src="https://github.com/user-attachments/assets/6cb70a9e-1978-490d-900f-3d3efda74be9" />

---
My AI search results:
> Why it's **not** "OP_RETURN": OP_RETURN is an opcode used by everyday users to embed data into standard transactions. Miners don't need to use it to leave a note.
> 
> Why it's "arbitrary": The speaker was referring to the arbitrary data string that mining pools choose to write inside the Coinbase scriptSig (the very first transaction of a block that mints new Bitcoin). F2Pool modified this custom text string in the blocks they mined during that period, which is the "message" the hosts were discussing.

> The coinbase transaction (the very first transaction in a block, which mints the new Bitcoin and collects fees) has a scriptSig field. Because this field doesn't need to unlock any previous coins, miners are free to inject a small amount of arbitrary data into it. Historically, this is where Satoshi wrote the famous "The Times 03/Jan/2009 Chancellor on brink of second bailout for banks" message.

@0tuedon Could you help verify if it's `arbitrary` at timestamp [00:02:53]?

But we explained that back in that episode.
Because as far as I know, they didn't actually do any filtering.

> But it caused quite a lot of consternations and they backed out of that idea.

Reviewer: https://www.collinsdictionary.com/dictionary/english/consternation
https://dictionary.cambridge.org/dictionary/english/consternation

I agree with you that the speaker said it with "s". I think it's speaker's mistake because consternation seems like an **uncountable** noun.

We can ignore this, if you like. Thanks for verifying it

This is kind of the thing that Bitcoin, Bitcoin developers and journalists, maybe as well, sometimes have been like talking about and warning about for many years, and this is sort of the first time it's actually happening.
So there's actually a reason now for…

Sjors Provoost: 00:09:22

> If you go through, you know, this, if you just assume a slippery slope, which is always a problematic type of argument, but if you do, then the ultimate outcome of this is you're back to PayPal because all transactions have to be KYC'd in permission before they go into a block.

Reviewer: <img width="1324" height="222" alt="Image" src="https://github.com/user-attachments/assets/de7a042f-1cde-4ee3-8c05-fd01406cd453" />

---
> "In permission" isn't a standard English idiom. Usually, something is permitted, requires permission, or operates on a permissioned network.

I think AI is correct.  "and" can sound like "in" since "**d**" can sound quite soft. Same as permission"e**d**"

Unless @0tuedon would like to verify and correct this, **we can ignore this**.

One of the first baby steps was the RPC called `submitpackage` and the idea was to send transactions to your own `mempool`, so not to your peers just your own `mempool`.
And I think half a year ago or so we covered how that worked.
You would give this RPC two transactions and it would just basically reject it because it would use the same logic as always, it would process them one by one.
So why was that added?
Well, because the next step makes it more useful.
Now with the 26.0 release, it will actually consider both of them and therefore include it in its own `mempool`.

Aaron van Wirdum: 00:35:43

Right.

Sjors Provoost: 00:35:44

But not really yet.

Aaron van Wirdum: 00:35:47

So this upgrade only applies to...
So if I make two transactions, one of them is a low fee, one of them is a high fee, I can now send this transaction over RPC to my own node?
But that's it, right?

Sjors Provoost: 00:36:03

Yeah, and you could imagine maybe there would be a website of a miner where you can send it directly to the miner this way and they would use their own RPC.
That's not the future obviously, but you could see some use even of this primitive feature.

Aaron van Wirdum: 00:36:17

Right.

> So there's now something in Bitcoin Core, it's a baby step, it's a step towards packet relay, right?

Reviewer: ```suggestion
So there's now something in Bitcoin Core, it's a baby step, it's a step towards package relay, right?
```

Reviewer: <img width="1958" height="274" alt="Image" src="https://github.com/user-attachments/assets/ae488c46-a0cb-46e8-b1f0-6e4752c01573" />

---
It sounds more like "package" to me, but not very clear. And `package relay `was mentioned several times in other places.

Also, this is a Bitcoin technical term. I believe @0tuedon would like it to be correct for their "knowledge base" data.

https://bitcoinops.org/en/topics/package-relay/
https://bips.dev/331/
